File: frontend/src/api/consultations.ts
import client from './client';
import type { StartConsultResponse, Message } from '../types';

export function startConsultation(doctorId: number, symptom: string) {
  return client
    .post<StartConsultResponse>('/consultations/start', { doctor_id: doctorId, symptom })
    .then((r) => r.data);
}

export function sendMessage(sessionId: number, content: string) {
  return client
    .post<{ reply: string; session_id: number }>(`/consultations/${sessionId}/message`, { content })
    .then((r) => r.data);
}

export function fetchMessages(sessionId: number) {
  return client.get<Message[]>(`/consultations/${sessionId}/messages`).then((r) => r.data);
}

File: backend/app/routers/consultations.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from app.database import get_db
from app.middleware.auth import require_role
from app.schemas.consultation import SessionCreate, MessageCreate, SessionResponse, MessageResponse
from app.services import consultation_service
from app.models.session import Session, Message

router = APIRouter(
    prefix="/api",
    tags=["consultations"],
    dependencies=[Depends(require_role("super_admin"))],
)


@router.post("/consultations/start")
def start_consultation(data: SessionCreate, db: Session = Depends(get_db)):
    session, doctor, reply = consultation_service.start_session(
        db, doctor_id=data.doctor_id, symptom=data.symptom, patient_name=data.patient_name
    )
    if not session:
        raise HTTPException(status_code=400, detail=reply)
    return {
        "session_id": session.id,
        "doctor": {"id": doctor.id, "name": doctor.name, "department": doctor.department, "avatar_url": doctor.avatar_url},
        "greeting": reply,
    }


@router.post("/consultations/{session_id}/message")
def send_message(session_id: int, data: MessageCreate, db: Session = Depends(get_db)):
    reply, error = consultation_service.send_message(db, session_id, data.content)
    if error:
        raise HTTPException(status_code=400, detail=error)
    return {"reply": reply, "session_id": session_id}


@router.get("/consultations/{session_id}/messages", response_model=list[MessageResponse])
def get_messages(session_id: int, db: Session = Depends(get_db)):
    return consultation_service.get_session_messages(db, session_id)


@router.get("/sessions", response_model=list[SessionResponse])
def list_sessions(doctor_id: int = None, db: Session = Depends(get_db)):
    q = db.query(Session)
    if doctor_id:
        q = q.filter(Session.doctor_id == doctor_id)
    return q.order_by(Session.created_at.desc()).all()
